internal/pkg/cli/svc_deploy.go

```diff
@@ -309,26 +314,67 @@ func (o *deploySvcOpts) pushToECRRepo() error {
 	return o.docker.Push(uri, o.ImageTag)
 }
 
-func (o *deploySvcOpts) getDockerfilePath() (string, error) {
-	type dfPath interface {
+type dockerParams struct {
+	context    string
+	dockerfile string
+	args       map[string]string
+}
+
+// toDockerBuildArguments fills in the fields contained in the image/build tag in the manifest.
+func (d *dockerParams) toDockerBuildArguments() docker.BuildArguments {
+	return docker.BuildArguments{
+		Context:    d.context,
+		Dockerfile: d.dockerfile,
+		Args:       d.args,
+	}
+}
+
+func (o *deploySvcOpts) getDockerfile() (*dockerParams, error) {
+	type dfContext interface {
+		DockerfileContext() string
 		DockerfilePath() string
+		DockerArgs() map[string]string
 	}
 
 	manifestBytes, err := o.ws.ReadServiceManifest(o.Name)
 	if err != nil {
-		return "", fmt.Errorf("read manifest file %s: %w", o.Name, err)
+		return nil, fmt.Errorf("read manifest file %s: %w", o.Name, err)
 	}
-
-	svc, err := manifest.UnmarshalService(manifestBytes)
+	svc, err := o.unmarshal(manifestBytes)
 	if err != nil {
-		return "", fmt.Errorf("unmarshal svc manifest: %w", err)
+		return nil, fmt.Errorf("unmarshal service %s manifest: %w", o.Name, err)
 	}
-
-	mf, ok := svc.(dfPath)
+	mf, ok := svc.(dfContext)
 	if !ok {
-		return "", fmt.Errorf("service %s does not have a dockerfile path", o.Name)
+		return nil, fmt.Errorf("service %s does not have required methods DockerArgs, DockerfilePath, and DockerfileContext", o.Name)
 	}
-	return mf.DockerfilePath(), nil
+	copilotDir, err := o.ws.CopilotDirPath()
+	if err != nil {
+		return nil, fmt.Errorf("get copilot directory: %w", err)
+	}
+	wsRoot := filepath.Dir(copilotDir)
+
+	dfPath := filepath.Join(wsRoot, mf.DockerfilePath())
+
+	dfArgs := mf.DockerArgs()
+
+	ctx := mf.DockerfileContext()
+	// Nonempty context field means we should join with the ws root and return that
+	if ctx != "" {
+		return &dockerParams{
+			dockerfile: dfPath,
+			context:    filepath.Join(wsRoot, mf.DockerfileContext()),
```

Comment:
so does mean that we assume the context is always relative to the root of the repo, is it possible that they enter an abs path? What would happen then? (btw I think I'm fine with this, we'll just call it out explicitly in the docs)

Reply:
Great question. I think what will happen is that docker will throw an error. We could catch this with the following: 
```go
ctx := mf.DockerfileContext()
if filepath.IsAbs(ctx) {
    rel, err := filepath.Rel(wsRoot, ctx) 
    if err != nil {
        return nil, err
    }
    ctx = rel
}   
return docker.BuildArguments{
    Context: filepath.Join(wsRoot, ctx)
    ...
}


// wsRoot is an absolute path, so we can get the 
// relative path between 2 absolutes with filepath.Rel
// ex filepath.Rel("/wsRoot", "/wsRoot/context") -> "context"
// filepath.Join("/wsRoot", "context") -> "/wsRoot/context"
```

Reply:
Yeah agree it might be overkill...I think it makes sense to users if we explain it well in our wiki/doc that users need to use the context that is relative to the copilot root.

Reply:
yup yup makes sense to me. People seem to get that the path in the "manifest" file is from the root of the repo so we can keep it as is